### ISSUE TITLE
full pathnames

### ISSUE THREAD

**Issue by tknuth** [Factual/drake#174]
I'm not sure whether this is a real issue or a problem of my configuration.

When using drake, the actions to be executed are shown with their full pathname. Is it somehow possible to display the paths only relative to the current working dir?

I run drake in "/Users/myname/dir1/dir2/dir3/dir4":

/Users/myname/dir1/dir2/dir3/dir4/././data/file.csv1 <- /Users/myname/dir1/dir2/dir3/dir4/././data/file2.csv

I'd rather want:

file1.csv <- file2.csv

What's happening there?

**Comment by dirtyvagabond** [comment]
@amalloy my memory is vague on this, but I *thought* we used to show a shorter version of the path. And it does seem a bit of clutter. you know anything about this?

**Comment by amalloy** [comment]
I don't see any code that looks like it tries to shorten that output. You can run `drake --base=.`, if you're in the same directory as the drakefile, and then you get shorter paths, but the default base is absolute (to avoid problems with relative paths), so you get absolute-paths in the confirmation dialog as well. We could easily enough strip out the base, I think, but then there's perhaps some risk of being unclear? What do you think?

**Comment by tknuth** [comment]
I'm not sure whether you actually ask me or your colleague, so I will state my opinion for the former case :-)

I perceive the current setting as rather cluttered. I have multiple processes (of course, this is why you use drake ...) and the screen becomes unreadable when the long paths are printed again and again.

Here are my suggestions:

- show relative instead of absolute paths (= omit all path components which are shared by the Drakefile and the workflow components)
- offer an option to switch to absolute paths
- optional: When starting the drake command, show the directory to which relative paths relate to

I prefer readability over absolute paths.

Whatever you decide: Thank you for your nice work on this handy software!

**Comment by amalloy** [comment]
So, I could apply d706f587, which makes the confirmation dialog less cluttered while not changing anything else, but I'm a little wary of doing so, since the code is littered with comments like "we shouldn't just not show the base prefix, because it could be ambiguous". 

I don't quite see how it would be ambiguous, personally, since I can't think of any case where the version without a BASE prefix could have a name clash but adding the BASE prefix removes the conflict. But I am so fond of bringing up [Chesterton's fence](http://en.wikipedia.org/wiki/Wikipedia:Chesterton%27s_fence), and this seems like a fence whose purpose I don't understand yet. 

@dirtyvagabond, do you have any idea why these comments from @aboytsov warn against removing the BASE prefix?

**Comment by dirtyvagabond** [comment]
thanks @amalloy for looking at this. mind pointing to the specific warnings?

**Comment by amalloy** [comment]
@dirtyvagabond https://github.com/Factual/drake/blob/develop/src/drake/core.clj#L57-L68 https://github.com/Factual/drake/blob/develop/src/drake/core.clj#L272-L276 were the two I noticed. Maybe there are more, I dunno.

**Comment by aboytsov** [comment]
I don't think we can simply omit $BASE when displaying the filenames since strictly speaking $BASE can be redefined in the workflow. Also, see #43 for more thoughts on BASE.

We can certainly show paths relative to CWD, at least for local file systems, and have an option to show absolute paths instead. 

Another options is slightly more cool and more involved. Even though BASE can change, it is defined for every step (that would also apply if #43 was implemented). We can use that and display BASE for each step, e.g.:

```
data/file.csv1 <- data/file2.csv [BASE:/Users/myname/dir1/dir2/dir3/dir4/]
```

If we synchronize on printing the steps (as we should), we can then only include this suffix when BASE changes. And we can omit this suffix if none of the filenames use BASE (they could be absolute filenames, too).

**Comment by pjakobsen** [comment]
--base is good, but it would be nice to be able to set that in some configuration file since the philosophy drake is to return a workflow after many months off or to share. One would sure forget the --base option after 6 months doing something else. Also, one shouldn't have to include a README with "HEY don't forget to set these options....etc." for users that share code
